src/test/java/uk/gov/pay/api/filter/ratelimit/RedisRateLimiterTest.java

```diff
@@ -0,0 +1,89 @@
+package uk.gov.pay.api.filter.ratelimit;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.JedisPool;
+
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class RedisRateLimiterTest {
+
+    @Mock
+    JedisPool jedisPool;
+    @Mock
+    Jedis jedis;
```

Comment:
Using mocks is discouraged. Can we use https://github.com/50onRed/mock-jedis which has real implementations of MockJedisPool and Jedis please.
Setting up expectations is prone to error and as commented below, I don't see how rateLimiterSetTo_2CallsPerSecond_shouldAllowMaking3CallsWithinTheAllowedTimeWithSameKey should pass. But when I use the MockJedisPool and MockJedis classes the test fails.